rust/src/lib/ifaces/inter_ifaces_controller.rs

```diff
@@ -157,6 +157,59 @@ impl MergedInterfaces {
         Ok(())
     }
 
+    // If port name reference should be check kernel name first, then fallback
+    // to profile name.
+    pub(crate) fn resolve_port_name_ref(&mut self) -> Result<(), NmstateError> {
+        let mut profile_2_kernel: HashMap<String, String> = HashMap::new();
+        for iface in self.kernel_ifaces.values() {
+            let base_iface = iface.merged.base_iface();
+            if let Some(profile_name) = base_iface.profile_name.as_deref() {
+                if profile_name != base_iface.name.as_str() {
+                    profile_2_kernel.insert(
+                        profile_name.to_string(),
+                        base_iface.name.to_string(),
+                    );
+                }
+            }
+        }
```

Comment:
Probably you can use a `HashMap<&str, &str>` here?

Reply:
We will change `des_iface`, `for_apply` and `for_verify` of `self`, so we have to own the data instead of referencing.